lib/cretonne/src/dominator_tree.rs

```diff
@@ -323,6 +329,72 @@ impl DominatorTree {
     }
 }
 
+impl DominatorTree {
+    /// When splitting an `Ebb` using `Layout::split_ebb`, you can use this method to update
+    /// the dominator tree locally rather than recomputing it.
+    ///
+    /// `old_ebb` is the `Ebb` before splitting, and `new_ebb` is the `Ebb` which now contains
+    /// the second half of `old_ebb`.
+    pub fn recompute_split_ebb(&mut self, old_ebb: Ebb, new_ebb: Ebb, split_jump_inst: Inst) {
+        let old_ebb_postorder_index =
+            self.postorder
+            .as_slice()
+            // We use the RPO comparison on the postorder list so we invert the operands of the
+            // comparison
+            .binary_search_by(|probe| self.rpo_cmp_ebb(old_ebb,*probe,))
+            .expect("the old ebb is not declared to the dominator tree");
```

Comment:
If `ebb` is unreachable then it stays unreachable after splitting, and `new_ebb` unreachable too. I'll deal with this case.